### PR TITLE
Bug/dsg 799/data path logic

### PR DESCRIPTION
**Pull request by jcsuarezm**
## Description
The follwing PR adds logic for managing dataPaths for CustomerProvidedDocuments (CPD). Each time a CPD is created or modified, the dataPath to the existing schema is validated to be correct, if needed this path is created, when documents are created via a data JSON. 

The affected endpoints are: 
1. PUT - /api/v1/transactions/:id
2. PUT - /api/v1/transactions/:transactionId/documents/:documentId

## Motivation and Context
There was a modification to this field, which was previously a schemaKey only, more validations are now needed. 

## Areas for Focus
Please focus on possible edge cases such as objects embedded into other objects, lists or other aspects. Validate the logic to get and set the dotted path is correct. 

## How Has This Been Tested?
Extensive unit tests were added. This was also tested via Postman (see below). 

## Screenshots (if appropriate):
1. Successful document creation: 
image

2. Document creation fails due to validation: 
image

3. Transaction update succeeds with new document: 
image

4. Non document validation prevents operation from occurring: 

image

## Types of changes

Patch
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [x] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

File: service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/Transaction.java [hunk 1]
```diff
@@ -66,6 +66,7 @@ public class Transaction implements DynamicEntityContainer {
             nullable = false,
             insertable = false,
             updatable = false)
+    @Setter
```
Comment (jgnuvalence) [inline service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/Transaction.java:69]: left a comment regarding this in TramnsactionService line 499

**Review by jgnuvalence — COMMENTED** [review]
this is working for me, I just left a couple of comments

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java [hunk 2]
```diff
@@ -161,53 +162,82 @@ public Transaction updateTransactionFromPartialUpdate(
...
 
-        Set<UUID> documentsToRemove =
-                existingDocuments.stream()
-                        .filter(ed -> !documentProperties.contains(ed))
-                        .collect(Collectors.toSet());
+        return mapUuidPath;
+    }
 
-        transaction.getCustomerProvidedDocuments().addAll(documentsToAdd);
+    private Object manageUnknownObjects(
```
Comment (JayStGelais) [inline service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java:181]: I think this method name is kind of confusing. It doesn't tell me what this method does.

File: service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/Transaction.java (hunk 1 above)
Comment (jcsuarezm) [reply service/src/main/java/io/nuvalence/workmanager/service/domain/transaction/Transaction.java]: The setter has been removed.

File: service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java (hunk 2 above)
Comment (jcsuarezm) [reply service/src/main/java/io/nuvalence/workmanager/service/service/TransactionService.java]: Thank you! The method has been refactored to better describe its purpose.

**Review by jgnuvalence — APPROVED** [review]
